extensions/client/kyuubi-spark-client/src/main/scala/org/apache/kyuubi/plugin/client/KyuubiSparkClientExtension.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.plugin.client
+
+import org.apache.spark.sql.SparkSessionExtensions
+import org.apache.spark.sql.jdbc.JdbcDialects
+
+import org.apache.kyuubi.plugin.client.dialect.HiveDialect
+
+class KyuubiSparkClientExtension extends (SparkSessionExtensions => Unit) {
+  override def apply(extensions: SparkSessionExtensions): Unit = {
+    // register hive jdbc dialect
+    JdbcDialects.registerDialect(HiveDialect)
```

Comment:
How about adding a switch? e.g.`spark.sql.jdbc.KyuubiHiveDialect.enabled`

Reply:
Good, and maybe we should leave it to `true` by default ?

Reply:
yea